xapian-core/cluster/cluster.cc

```diff
@@ -0,0 +1,258 @@
+/** @file cluster.cc
+ *  @brief Cluster API
+ */
+/* Copyright (C) 2010 Richard Boulton
+ * Copyright (C) 2016 Richhiey Thomas
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include "xapian/cluster.h"
+
+#include <xapian/error.h>
+#include <api/termlist.h>
+#include <debuglog.h>
+
+#include <vector>
+#include <map>
+
+using namespace Xapian;
+using namespace std;
+
+MSetDocumentSource::MSetDocumentSource(const MSet &mset_)
+: mset(mset_), maxitems(mset_.size()), index(0)
```

Comment:
We usually offset initialisers by 4 characters. (I say "usually", because I think some older bits of the code don't.) In my opinion it makes it slightly easier to scan through and see where the signatures and bodies are.